validator/lib/jsonapi/validator/resource.rb

```diff
@@ -0,0 +1,123 @@
+require 'jsonapi/validator/document'
+require 'jsonapi/validator/relationship'
+
+module JSONAPI
+  module Validator
+    class Resource
```

Comment:
When it's less about gem dependencies and more about a wasteful intermediate step then I'm with you. I would *personally* choose to keep the wastefulness and use some already-existing gems, but I can see how this would be less than ideal and others would prefer to solve the root problem. If nothing else, I think the conversation has re-emphasized making this a separate gem was a good step to take 👍